src/core/opcua-iiot-core-response.ts

```diff
@@ -59,7 +59,7 @@ const analyzeBrowserResults = function (node: Node, payload: BrowserPayload) {
 }
 
 const analyzeCrawlerResults = function (node: Node, payload: CrawlerPayload) {
-  handlePayloadStatusCode(node, payload.value, payload as AnyPayload)
+  handlePayloadStatusCode(node, payload.value as StatusInput[], payload as AnyPayload)
```

Comment:
This wouldn't work because `CrawlerPayload` is not of type `StatusInput[]`. A good solution could be a broader input type of the `handlePayloadStatusCode` function. So instead of only allowing:

```ts
const handlePayloadStatusCode = function (node: Node, statusInputs: StatusInput | StatusInput[], payload: AnyPayload) {
  ...
}
```

it could have the signature

```ts
const handlePayloadStatusCode = function<T extends Record<any, any>>(
  node: Node,
  statusInputs: T | T[],
  payload: any
) {
   ...
}
```

And inside the function you check the `statusInputs` parameter if it really is a statusInput or not with a guard function:

```ts
const isStatusInput = (val: Record<any, any>): val is StatusInput => {
  if ("statusCode" in val && val instanceof StatusCode) {
    return true;
  }
  if ("statusCodes" in val && Array.isArray(val.statusCodes)) {
    return val.every((item) => item instanceof StatusCode);
  }
  return false;
};
```

This way you have the type safety from TypeScript (which was removed by using `as`) and you're able to check every `object` for which you suspect that it could contain an unexpected `statusCode` or `statusCodes` field.